### PR TITLE
Benchmark fixes

### PR DESCRIPTION
**Pull request by kciesielski**
This PR aims to fix threading problems in consumer benchmarks + some other minor issues. Producer benchmarks will be fixed/finished in a separate PR.

### CODE REVIEW

File: benchmarks/src/main/scala/akka.kafka.benchmarks/KafkaConsumerBenchmarks.scala [hunk 1]
```diff
@@ -71,37 +71,40 @@ object KafkaConsumerBenchmarks extends LazyLogging {
 
     var lastProcessedOffset = 0L
     var accumulatedMsgCount = 0L
-    val semaphore = new Semaphore(1)
+    val commitInProgress = new AtomicBoolean(false)
```
Comment (patriknw) [inline benchmarks/src/main/scala/akka.kafka.benchmarks/KafkaConsumerBenchmarks.scala]: There is only one thread involved, as far as I understand the Kafka threading model. Callbacks are invoked by the thread invoking `poll`. That would mean that there is no need for atomic.

Comment (kciesielski) [inline benchmarks/src/main/scala/akka.kafka.benchmarks/KafkaConsumerBenchmarks.scala]: Ok, fixed

**Comment by patriknw** [comment]
LGTM

**Comment by patriknw** [comment]
I have restarted the Travis build. We have some issue with IntegrationSpec, but that is unrelated to this. Issue created: https://github.com/akka/reactive-kafka/issues/203

**Comment by patriknw** [comment]
@kciesielski rebase and then merge

File: benchmarks/src/main/scala/akka.kafka.benchmarks/KafkaConsumerBenchmarks.scala [hunk 2]
```diff
@@ -71,39 +69,42 @@ object KafkaConsumerBenchmarks extends LazyLogging {
...
           lastProcessedOffset = record.offset()
-          if (accumulatedMsgCount >= batchSize || noCommitInProgress)
-            doCommit()
+          if (accumulatedMsgCount >= batchSize) {
+            if (!commitInProgress) {
+              commitInProgress = true
+              doCommit()
+            }
+            else // previous commit still in progress
+              consumer.pause(assignment)
```
Comment (drewhk) [inline benchmarks/src/main/scala/akka.kafka.benchmarks/KafkaConsumerBenchmarks.scala]: yep, now it does the same as the reactive version

**Comment by drewhk** [comment]
Looks good!

Will you update the benchmark findings eventually?

**Comment by kciesielski** [comment]
@drewhk Yes, I'll update the results.

**Pull request merged**